documentation/src/docs/asciidoc/user-guide/writing-tests.adoc

```diff
@@ -367,6 +367,100 @@ NOTE: As of JUnit Jupiter 5.4, it is also possible to use methods from JUnit 4's
 `AssumptionViolatedException` to signal that a test should be aborted instead of marked
 as a failure.
 
+[[writing-tests-exceptions]]
+=== Exception Handling
+
+JUnit Jupiter provides robust support for handling test exceptions. This includes the
+built-in mechanisms for managing test failures due to exceptions, the role of exceptions
+in implementing assertions, and how to specifically assert non-throwing conditions in code.
+
+[[writing-tests-exceptions-uncaught]]
+==== Uncaught Exceptions and Test Failures
+
+In JUnit Jupiter, if an exception is thrown and not caught within the test method, the
+framework will mark the test as failed.
+
+[source,java,indent=0]
+----
+include::{testDir}/example/exception/UncaughtExceptionHandlingDemo.java[tags=user_guide]
+----
+
+In the above example, the `testThatFailsDueToUncaughtException` method throws a
+`RuntimeException`. Since the exception is not caught within the test method, JUnit
+Jupiter will automatically fail the test.
+
+NOTE: It's important to clarify that specifying a `throws` clause in the test method
+has no effect on the outcome of the test. JUnit Jupiter does not interpret it as
+an expectation or assertion about what exceptions the test method should throw. A test
+fails only if an exception is thrown unexpectedly or if an assertion fails.
+
+[[writing-tests-exceptions-assertions]]
+==== Assertions Using Exceptions
+
+Assertions in JUnit Jupiter are implemented using exceptions. The framework provides
+a set of assertion methods in the `org.junit.jupiter.api.Assertions` class, which throw
+`AssertionError` when an assertion fails. This mechanism is a core aspect of how JUnit
+handles assertion failures as exceptions.
+
+NOTE: This failed assertion behaviour (i.e. throwing `AssertionError`) may be different
+for third party assertion libraries. See also: <<writing-tests-assertions-third-party>>.
+
+NOTE: JUnit Jupiter itself does not differentiate between failed assertions
+(`AssertionError`) and other types of exceptions. All uncaught exceptions lead to a test
+failure. However, Integrated Development Environments (IDEs) and other tools may
+distinguish between these two types of failures by checking whether the thrown exception
+is an instance of `AssertionError`.
+
+[[writing-tests-exceptions-expected]]
+==== Asserting Expected Exception Conditions
+
+JUnit Jupiter offers specialized assertions for testing that specific exceptions are
+thrown under expected conditions. The `assertThrows` and `assertThrowsExactly` assertions
+are critical tools for validating that your code responds correctly to error conditions
+by throwing the appropriate exceptions.
+
```

Comment:
Maybe it would be useful to mention that `assertThrows` and `assertThrowsExactly` return the thrown exception to allow performing additional assertions on it, such as checking its message?
And then maybe include an example for this in `ExceptionAssertionDemo` and `ExceptionAssertionExactDemo` (or adjust the existing ones)?

Reply:
Yes, I think it's worth mentioning that explicitly.

The `exceptionTesting()` example in the Assertions section already demonstrates how to check the exception message in the returned exception, but I agree that it's worth repeating that in the `assertThrows` and `assertThrowsExactly` examples in these new sections.